FindImportProfile/FindImportProfile.js

```diff
@@ -55,10 +58,12 @@ const FindImportProfile = ({
   const fetchAssociations = async (props, id) => {
     const { stripes: { okapi } } = props;
     const { url } = okapi;
-    const baseUrl = `${url}/data-import-profiles/profileAssociations/${id}/masters`;
+    const profileType = masterType === parentType ? 'masters' : 'details';
+    const queryProfileType = masterType === parentType ? 'detailType' : 'masterType';
+    const baseUrl = `${url}/data-import-profiles/profileAssociations/${id}/${profileType}`;
 
     const response = await fetch(
-      createUrl(baseUrl, { detailType: ASSOCIATION_TYPES[entityKey] }, false),
+      createUrl(baseUrl, { [queryProfileType]: ASSOCIATION_TYPES[entityKey] }, false),
       { headers: { ...createOkapiHeaders(okapi) } },
     );
```

Comment:
Why don't you want to split logic of request and component render in different files? It's just hard to understand what is happening here

Reply:
Thank you for catching this, split it into the `utils` file.